src/Propel/Common/Util/CartesianProduct.php

```diff
@@ -0,0 +1,43 @@
+<?php namespace Propel\Common\Util;
+
+class CartesianProduct
+{
+    public static function get($input) {
```

Comment:
There's a good opportunity to decrease the logical nesting here. If I get a moment I might propose an alternative that uses generators (since propel is now 7.1+).